fluster/image/raw_buffer.py

```diff
@@ -0,0 +1,361 @@
+        return self._buffer
+
+    @property
+    def pixels_to_ignore(self) -> PixelToIgnore:
+        """Returns the list of pixels to ignore in comparison"""
+        return self._pixels_to_ignore
+
+    @property
+    def buffer_type(self) -> Type:
+        """Returns the buffer type of the class"""
+        return self._buffer_type
+
+    @property
+    def width(self) -> int:
+        """Returns the width of the image stored in the buffer"""
+        return self._width
+
+    @property
+    def height(self) -> int:
+        """Returns the height of the image stored in the buffer"""
+        return self._height
+
+    @property
+    def frame_size(self) -> int:
+        """Returns the size of the frame stored in the buffer"""
+        return self._height * self._width
+
+    def _rgba(self, index: int) -> Tuple[int, int, int, int]:
+        """Returns the RGBA tuple from the indicated position"""
+        return (
+            self._buffer[index] & 0xFF,
+            self._buffer[index + 1] & 0xFF,
+            self._buffer[index + 2] & 0xFF,
+            self._buffer[index + 3] & 0xFF,
+        )
+
+    @staticmethod
+    def _rgba_to_yuv(rgba_value: Tuple[int, int, int, int]) -> Tuple[int, int, int]:
+        """It converts an RGBA tuple to another YUV420 tuple"""
+
+        def y_from_rgba(_r: int, _g: int, _b: int, _a: int) -> int:
+            return RawBuffer.clamp(((66 * _r + 129 * _g + 25 * _b + 128) >> 8) + 16)
+
+        def u_from_rgba(_r: int, _g: int, _b: int, _a: int) -> int:
+            return RawBuffer.clamp(((-38 * _r - 74 * _g + 112 * _b + 128) >> 8) + 128)
+
+        def v_from_rgba(_r: int, _g: int, _b: int, _a: int) -> int:
+            return RawBuffer.clamp(((112 * _r - 94 * _g - 18 * _b + 128) >> 8) + 128)
+
+        return (
+            y_from_rgba(*rgba_value),
+            u_from_rgba(*rgba_value),
+            v_from_rgba(*rgba_value),
+        )
+
+    def _yuv420_comparison(
+        self, target: "RawBuffer", deviations: Dict[int, int]
+    ) -> int:
+        """Comparison of two buffers of type YUV420"""
+        source_buffer: bytes = self.buffer
+        target_buffer: bytes = target.buffer
+        size_chroma: int = self.frame_size // 4
+        start_u: int = self.frame_size
+        start_v: int = self.frame_size + size_chroma
+        total_errors = 0
+
+        def check_deviation(source_pixel: int, target_pixel: int) -> int:
+            deviation = int(abs(source_pixel - target_pixel))
+            if deviation > self._YUV_TOLERANCE:
+                if i not in self._pixels_to_ignore:
+                    deviations[deviation] = (
+                        1 if deviation not in deviations else deviations[deviation] + 1
+                    )
+                    return 1
+            return 0
+
+        # First compare the y buffer
+        for i in range(self.frame_size):
+            total_errors += check_deviation(source_buffer[i], target_buffer[i])
+            if i < size_chroma:
```

Comment:
I do not understand you, I am making a loop that simultaneously goes through y, u, and v arrays, but since u and v arrays are smaller, I must control not to use an index that passes them.

Reply:
I understand that, but what is _deviations[i]_ representing? Is it the deviation of the pixel _i_? Because if so, then I see a problem.

Reply:
I'm not using pixels. I'm using bytes all the time.